sys/xtimer/xtimer.c

```diff
@@ -243,12 +243,16 @@ int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t timeout)
     xtimer_t t;
     mutex_thread_t mt = { mutex, (thread_t *)sched_active_thread, 0 };
 
-    if (timeout != 0) {
-        t.callback = _mutex_timeout;
-        t.arg = (void *)((mutex_thread_t *)&mt);
-        _xtimer_set64(&t, timeout, timeout >> 32);
+    /* timeout lower than XTIMER_BACKOFF might cause the code to spin rather
+     * than set it up for interrupt */
+    if (timeout <= XTIMER_BACKOFF) {
+        return (mutex_trylock(mutex) - 1);
```

Comment:
Let's say A has the mutex and B calls `xtimer_mutex_lock_timeout()` with `timeout <= XTIMER_BACKOFF`. Now A receives an interrupt that causes it to release the mutex. B could do its spin now and lock the mutex, right?